patch/kconfig-inclusions

```diff
@@ -126,6 +126,62 @@ CONFIG_SENSORS_EMC2305=m
 [marvell-arm64]
 CONFIG_MARVELL_AC5_WATCHDOG=m
 
+[pensando]
+CONFIG_ARCH_PENSANDO=y
+CONFIG_ARCH_PENSANDO_ELBA_SOC=y
+CONFIG_PENSANDO_SOC_CAPMEM_HUGEPAGE=y
+CONFIG_PENSANDO_SOC_PENFW=y
+CONFIG_EDAC_ELBA=y
+CONFIG_IRQ_PENSANDO=y
+CONFIG_MFD_PENSANDO_ELBASR=y
+CONFIG_RESET_ELBASR=y
+CONFIG_UIO_PCIEMAC=m
+CONFIG_UIO_PENMSI1=m
+CONFIG_UIO_PENGIC=m
+CONFIG_UIO_PENMSI=m
+CONFIG_ARM64_VA_BITS_39=y
+CONFIG_COMMON_CLK_CS2000_CP=y
+CONFIG_CONTEXT_TRACKING=y
+CONFIG_CRYPTO_MANAGER_DISABLE_TESTS=y
+CONFIG_DEFAULT_SECURITY_DAC=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_EMBEDDED=y
+CONFIG_EXT3_FS=y
+CONFIG_GENERIC_IRQ_DEBUGFS=y
+CONFIG_GPIO_DWAPB=y
+CONFIG_I2C_RD1173=y
+CONFIG_IP_PNP=y
+CONFIG_IP_PNP_BOOTP=y
+CONFIG_IP_PNP_DHCP=y
+CONFIG_IRQSOFF_TRACER=y
+CONFIG_KALLSYMS_ALL=y
+CONFIG_LEGACY_PTYS=y
+CONFIG_MEMTEST=y
+CONFIG_MMC_DEBUG=y
+CONFIG_MMC_SDHCI_CADENCE=m
+CONFIG_NO_HZ_FULL=y
+CONFIG_PENSANDO_SOC_CAPMEM_HUGEPAGE=y
+CONFIG_PENSANDO_SOC_PENFW=y
+CONFIG_PMBUS=y
+CONFIG_PREEMPTIRQ_TRACEPOINTS=y
+CONFIG_RCU_NOCB_CPU=y
+CONFIG_RING_BUFFER_ALLOW_SWAP=y
+CONFIG_RTC_DRV_PCF85363=y
+CONFIG_SCHED_TRACER=y
+CONFIG_SENSORS_LTC2978=y
+CONFIG_SENSORS_PMBUS=y
+CONFIG_SENSORS_TPS53679=y
+CONFIG_SERIAL_8250_DEPRECATED_OPTIONS=y
+CONFIG_SPI_CADENCE_QUADSPI=m
+CONFIG_SPI_DESIGNWARE=m
+CONFIG_SPI_DW_MMIO=m
+CONFIG_SQUASHFS_DECOMP_MULTI=y
+CONFIG_SQUASHFS_FILE_CACHE=y
+CONFIG_SYSFS_SYSCALL=y
```

Comment:
This is a deprecated config, why does this need to be enabled?

Reply:
This has been inherited from buildroot (on which pensando kernel is tested on). By removing this might have compatibility issues as mentioned in the config.

Reply:
From my understanding, the userspace and kernel that buildroot uses are different than what SONiC uses, so there are already differences. It looks like buildroot is for bringing up embedded systems easily, which explains why many of these config modules that are usually enabled as a module are instead compiled into the kernel.

Since this will be using SONiC userspace, I would strongly recommend having only platform-specific config options (such as architecture-related configs, enabling support for hardware, etc.) here. Several of these configs appear to be primarily for debugging or are not platform-specific (`CONFIG_LEGACY_PTYS`, `CONFIG_DEFAULT_SECURITY_DAC`, `CONFIG_CRYPTO_MANAGER_DISABLE_TESTS`, `CONFIG_SERIAL_8250_DEPRECATED_OPTIONS`, `CONFIG_SYSFS_SYSCALL`, `CONFIG_DEVTMPFS_MOUNT`).

Reply:
@saiarcot895 I don't understand why we can't have CONFIGS enabled for debugging ? It is upto the platform to decide.

Reply:
Also, this is not just SONiC userspace. "Full" pensando userspace (which is built in buildroot and which should be compatible with the corresponding kernel) runs in SONiC inside a docker. Please talk to @lguohan if u are not familiar with the architecture.

Reply:
userspace is sonic, pensando software is mainly running with docker. we need to get rid of legacy deprecated options since they are hard to maintain.

Reply:
Pensando SW (running in docker) is also userspace. We can check on legacy option but debugging option should be left to the platform (for ex: CONFIG_DEVTMPFS_MOUNT for rescue mode).